server/schedule/placement/rule_manager.go

```diff
@@ -318,10 +326,21 @@ func (m *RuleManager) GetRulesForApplyRange(start, end []byte) []*Rule {
 	return m.ruleList.getRulesForApplyRange(start, end)
 }
 
+func (m *RuleManager) checkAndAjustRules(rules []*Rule) {
+	if !m.opt.IsSwitchWitnessAllowed() {
+		for _, rule := range rules {
+			if rule.IsWitness {
+				rule.IsWitness = false
```

Comment:
When will it become true?

Reply:
Instead of directly modifying the rule, use the discussion method, PTAL again.